Dockerfile

```
@@ -1,22 +1,25 @@
-FROM python:3.11.3-slim-bullseye as docs
+FROM python:3.12.5-slim-bookworm as docs
 
 RUN set -x \
     && apt-get update \
     && apt-get install --no-install-recommends -y \
         build-essential git libcairo2-dev libfreetype6-dev \
-        libjpeg-dev libpng-dev libz-dev
+        libjpeg-dev libpng-dev libz-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
 RUN set -x \
     && python3 -m venv /opt/venv
 
 ENV PATH="/opt/venv/bin:${PATH}"
 
-RUN pip --no-cache-dir --disable-pip-version-check install \
-    --upgrade pip setuptools wheel
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1
+ENV PIP_NO_CACHE_DIR=1
```

Not sure if this makes much difference?	
```suggestion
ARG PIP_DISABLE_PIP_VERSION_CHECK=1
ARG PIP_NO_CACHE_DIR=1
```

* `ARG` - only available during the build
* `ENV` - also available inside the image

https://vsupalov.com/docker-arg-env-variable-guide/#arg-and-env
https://docs.docker.com/build/building/variables/